teps/0048-task-results-without-results.md

```diff
@@ -283,6 +284,35 @@ spec:
       value: $(tasks.propagate-image-name.results.image)
 ```
 
+#### Providing default values for optional `Results`
+
+Today, a `Task` can declare a `Result` but not produce the `Result`
+and it's execution will still be successful. If a subsequent `Task`
+attempts to use the `Result`, then it would fail (causing the `Pipeline` to fail).
+As such, a `Result` is optional when unused but required when used - this
+can be confusing to users when failures happen later in the `Pipeline`.
+Supporting default values for the optional `Results` would help prevent 
+unexpected failures, and improve our `Task` interface`.
+
+For example, the following `Task` will produce an optional
+result when the none of the `Steps` in the `Task` produces the
+results:
+
+```yaml
+...
+spec:
+  ...
+  results:
+    - name: revision
+      description: Revision which was checkout. (branch, tag, sha, ref, etc...)
+      default: "main"
+  steps:
+    - name: fetch-checkout
+      image: alpine/git:v2.26
+      workingDir: $(workspaces.source.path)
+      script: |
+        git clone https://github.com/tektoncd/community
```

Comment:
From my pov this is an error in the Task's logic. If I wrote this I would not want it to succeed - I made a mistake that could have bad consequences for downstream users of my Task.

Is there maybe a more real-world example to demonstrate the use-case?

Reply:
other examples are here: https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md#use-cases

and another one here for manual approval: https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md#summary

@vinamra28 - maybe we can reuse one of the the scenarios in the examples already in the TEP?